jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/HttpClientContinueTest.java

```diff
@@ -676,6 +678,56 @@ public void onComplete(Result result)
         assertTrue(latch.await(5, TimeUnit.SECONDS));
     }
 
+    @ParameterizedTest
+    @MethodSource("transportsNoFCGI")
+    public void test100ContinueThenTimeoutThenSendError(Transport transport) throws Exception
+    {
+        long idleTimeout = 1000;
+
+        CountDownLatch serverLatch = new CountDownLatch(1);
+        startServer(transport, new HttpServlet()
+        {
+            @Override
+            protected void service(HttpServletRequest request, HttpServletResponse response) throws IOException
+            {
+                // Send the 100 Continue.
+                ServletInputStream input = request.getInputStream();
+                try
+                {
+                    // Echo the content.
+                    IO.copy(input, response.getOutputStream());
+                }
+                catch (IOException x)
+                {
+                    // The copy failed b/c of idle timeout, time to try
+                    // to send an error which should have no effect.
+                    response.sendError(HttpStatus.IM_A_TEAPOT_418);
+                    serverLatch.countDown();
+                }
+            }
+        });
+        startClient(transport, httpClient -> httpClient.setIdleTimeout(idleTimeout));
+
+        AsyncRequestContent requestContent = new AsyncRequestContent();
+        requestContent.write(ByteBuffer.wrap(new byte[512]), Callback.NOOP);
+        CountDownLatch clientLatch = new CountDownLatch(1);
+        client.newRequest(newURI(transport))
+            .headers(headers -> headers.put(HttpHeader.EXPECT, HttpHeaderValue.CONTINUE.asString()))
+            .body(requestContent)
+            .send(result ->
+            {
+                if (result.isFailed() && result.getResponse().getStatus() == HttpStatus.CONTINUE_100)
+                    clientLatch.countDown();
+            });
+
+        // Wait more than the idle timeout to break the connection.
+        Thread.sleep(2 * idleTimeout);
+
+        assertTrue(serverLatch.await(5, TimeUnit.SECONDS));
+        assertTrue(clientLatch.await(5, TimeUnit.SECONDS));
+        assertThat(jobFailures, empty());
```

Comment:
Why this last assert? Seems way too far away in abstractions layers to be tested here.

Reply:
This was the only half-sane way I found to create a reliable reproducer for the problem initially reported in #10357. I added this test as part of PR #10814 upon which this PR is based.

I detailed the actual problem here: https://github.com/jetty/jetty.project/issues/10357#issuecomment-1786824936 and discussed with @gregw here: https://github.com/jetty/jetty.project/pull/10801#discussion_r1376991169

It's not directly related to the transient failures, but it's part of the family of problems for which transient failures are a piece of the puzzle.

Reply:
After discussing this, we agreed to remove the assertion.